### PR TITLE
VZ-4272 skipping image check for pods matching skip image prefix

### PR DESCRIPTION
**Pull request by pshirodkar-orcl**
Signed-off-by: Prasad Shirodkar <prasad.shirodkar@oracle.com>

# Description

Please include a summary of the change and which issue is fixed.  If there are any dependencies, for example on a PR in another repository, please list those.

Fixes VZ-9999

# Checklist 

As the author of this PR, I have:

- [ ] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [x] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: tests/e2e/registry/registry_test.go [hunk 1]
```diff
@@ -76,14 +77,25 @@ var _ = t.Describe("Private Registry Verification", Label("f:platform-lcm.privat
...
-								fmt.Sprintf("FAIL: The image for the pod %s in initContainers, doesn't starts with expected registry URL prefix %s, image name %s", pod.Name, registry, pod.Spec.InitContainers[k].Image))
+							assertImageRegistryURLOfPod(pod.Name, pod.Spec.InitContainers[k].Image, imagePrefix, "initContainers")
 						}
 					}
 				}
 			})
 	})
+
+func assertImageRegistryURLOfPod(podName string, image string, imagePrefix string, containersStr string) {
+	if strings.TrimSpace(skipImagePrefix) != "" {
```
Comment (michael-cico) [inline tests/e2e/registry/registry_test.go:91]: what is the skip prefix thing for?  Not sure I follow what it is.

Comment (pshirodkar-orcl) [reply tests/e2e/registry/registry_test.go:91]: Please see comment in slack.

File: tests/e2e/registry/registry_test.go [hunk 2]
```diff
@@ -76,14 +77,25 @@ var _ = t.Describe("Private Registry Verification", Label("f:platform-lcm.privat
...
+		if !strings.HasPrefix(image, skipImagePrefix) {
+			// Assertion only when the skip image prefix is set and excluding the images check matching the skip image prefix
+			Expect(strings.HasPrefix(image, imagePrefix)).To(BeTrue(),
```
Comment (michael-cico) [inline tests/e2e/registry/registry_test.go:94]: It seems like we're repeating this `Expect` block twice...I think what we essentially want is 

```
if strings.TrimSpace(skipImagePrefix) != "" && strings.HasPrefix(image, skipImagePrefix) {
   return 
}
Expect(strings.HasPrefix(image, imagePrefix)).To(BeTrue(),
			fmt.Sprintf("FAIL: The image for the pod %s in %s, doesn't starts with expected registry URL prefix %s, image name %s", podName, containersStr, imagePrefix, image))
```

Comment (michael-cico) [reply tests/e2e/registry/registry_test.go:94]: You don't have to hold up this merge for that, this is a nit.  but might simplify the code a bit on a second pass.

**Review by michael-cico — APPROVED** [review]

**Comment by pshirodkar-orcl** [comment]
This PR is no longer required.

**Pull request closed without merging**